### PR TITLE
Make the dialog boxes for kubernetes startup/shutdown errors friendlier

### PR DESCRIPTION
**Pull request by ericpromislow**
* Relay the error text from the worker process back to the main process

* Ensure the context is correct

* Replace undefined error codes with 0

### CODE REVIEW

**Review by mook-as — APPROVED** [review]

File: background.js [hunk 1]
```diff
@@ -126,8 +131,19 @@ ipcMain.on('k8s-restart', async (event) => {
   }
 });
 
-function startfailed(code) {
-  dialog.showErrorBox("Error Starting Kuberentes", "Kubernetes was unable to start with the following exit code: " + code);
+function startfailed(payload) {
```
[the remark this answers is not in the document]
Comment (ericpromislow) [reply background.js:134]: `handleFailure` would be a better name.

[the remark this answers is not in the document]
Comment (mattfarina) [reply background.js]: We can rename in a follow-up. I like `handleFailure` better.

[the remark this answers is not in the document]
Comment (mattfarina) [reply background.js]: Oh, I see it already went in. Nevermind.

**Review by mattfarina — APPROVED** [review]

**Comment by mattfarina** [comment]
@ericpromislow Can you fix the merge conflict

**Comment by mattfarina** [comment]
Note, this is one of many PRs that will be working to better error handling.

Once piece of feedback for this one is that suggested minikube commands can't be run as is due to the minikube home being set to a non-default location.

**Pull request merged**